File: android/src/main/java/com/margelo/nitro/visioncameramlkit/BarcodeScannerOptions+toDomain.kt
package com.margelo.nitro.visioncameramlkit

import com.visioncameramlkit.domain.models.BarcodeScanningOptions
import com.visioncameramlkit.domain.models.ImagePreprocessingOptions

internal fun BarcodeScannerOptions.toDomainBarcodeScanningOptions(): BarcodeScanningOptions =
  BarcodeScanningOptions(
    formats = formats.toDomainBarcodeFormats(),
    enableAllPotentialBarcodes = enableAllPotentialBarcodes ?: false,
    invertColors = invertColors ?: false,
    scaleFactor = scaleFactor?.toFloat() ?: 1.0f,
  )

internal fun BarcodeScannerOptions.toImagePreprocessingOptions(): ImagePreprocessingOptions =
  ImagePreprocessingOptions(
    invertColors = invertColors ?: false,
    scaleFactor = scaleFactor?.toFloat() ?: 1.0f,
  )

File: ios/Bridge/Hybrid/BarcodeScannerOptions+toDomain.swift
import Foundation

extension BarcodeScannerOptions {
  func toDomainBarcodeScanningOptions() -> BarcodeScanningOptions {
    return BarcodeScanningOptions(
      formats: formats?.map { $0.toDomainBarcodeFormat() } ?? [],
      enableAllPotentialBarcodes: enableAllPotentialBarcodes ?? false,
      invertColors: invertColors ?? false,
      scaleFactor: CGFloat(scaleFactor ?? 1.0)
    )
  }

  func toImagePreprocessingOptions() -> ImagePreprocessingOptions {
    return ImagePreprocessingOptions(
      invertColors: invertColors ?? false,
      scaleFactor: CGFloat(scaleFactor ?? 1.0),
      orientation: orientation?.toDomainOrientation(),
      roi: roi?.toDomainRegionOfInterest()
    )
  }
}
